examples/plot_intelex.py

```diff
@@ -0,0 +1,273 @@
+optimization survived being saved in the Hub.
+"""
+
+# %%
+# Imports
+# =======
+# First, we import everything required for the rest of this document.
+
+import os
+import pickle
+from pathlib import Path
+from tempfile import mkdtemp
+from time import perf_counter
+from uuid import uuid4
+
+from huggingface_hub import delete_repo, whoami
+from sklearn.datasets import make_classification
+from sklearn.metrics import log_loss
+from sklearn.model_selection import train_test_split
+from sklearnex import patch_sklearn
+
+from skops import card, hub_utils
+
+# %%
+# Data
+# ====
+# Next, we create some generic data. A dataset of 50k rows x 15 columns is big enough
+# to showcase a performance gain from using ``sklearnex``. Generally speaking,
+# larger datasets will benefit more from the ``sklearnex`` optimizations. More
+# details can be found in the official
+# `README <https://github.com/intel/scikit-learn-intelex/blob/master/README.md>`_.
+X, y = make_classification(
+    n_samples=50_000,
+    n_features=15,
+    n_informative=15,
+    n_redundant=0,
+    n_clusters_per_class=1,
+    shuffle=False,
+    random_state=42,
+)
+X_train, X_test, y_train, y_test = train_test_split(
+    X, y, test_size=0.3, random_state=42
+)
+
+# %%
+# Training the stock model
+# ========================
+# Now we can train a stock Scikit-learn
+# :class:`sklearn.neighbors.KNeighborsClassifier`
+
+from sklearn.neighbors import KNeighborsClassifier
+
+clf = KNeighborsClassifier(3)
+clf.fit(X_train, y_train)
+
+# %%
+# Training the optimized model
+# ============================
+# We apply ``patch_sklearn()`` and reimport the model to load the patched
+# version. A message is shown, telling us that Intel(R) Extension for
+# Scikit-learn* has been enabled.
+patch_sklearn()
+from sklearn.neighbors import KNeighborsClassifier
+
+clf_opt = KNeighborsClassifier(3)
+clf_opt.fit(X_train, y_train)
+
+# %%
+# Comparing inference times
+# =========================
+# Now to the interesting part: We measure the execution time of
+# ``predict_proba()`` for the two models.
+
+start = perf_counter()
+prediction = clf.predict_proba(X_test)
+t_stock = perf_counter() - start
+
+accuracy = log_loss(y_test, prediction)
+print(
+    f"[stock scikit-learn] Inference took t_stock = {t_stock:.2f}s and achieved"
+    f" {accuracy:.1%} accuracy"
+)
+
+start = perf_counter()
+prediction = clf_opt.predict_proba(X_test)
+t_opt = perf_counter() - start
+
+accuracy = log_loss(y_test, prediction)
+print(
+    f"[sklearnex] Inference took t_opt = {t_opt:.2f}s and achieved"
+    f" {accuracy:.1%} accuracy"
+)
+
+print(f"t_stock / t_opt = {t_stock/t_opt:.1f}")
+
+# %%
+# We see that inference using ``sklearnex`` is a lot faster while producing the
+# same result.
+
+# %%
+# Save and upload the models
+# ==========================
+# Let's save all required files to disk and initialize Hugging Face Model Hub
+# repositories.
```

Comment:
It could be worth mentioning that sklearnex will be used on the Hub, e.g. for the Inference API. This might not be obvious to everyone.